CMakeLists.txt

```diff
@@ -532,6 +544,21 @@ if (HDF4_BUILD_XDR_LIB)
   endif ()
 endif ()
 
+#-----------------------------------------------------------------------------
+# Option to build documentation
+#-----------------------------------------------------------------------------
+option (HDF4_BUILD_DOC "Build documentation" OFF)
+if (HDF4_BUILD_DOC AND EXISTS "${HDF4_DOXYGEN_DIR}" AND IS_DIRECTORY "${HDF4_DOXYGEN_DIR}")
+# check if Doxygen is installed
+  find_package(Doxygen)
+  if (DOXYGEN_FOUND)
+    message(STATUS "Doxygen version: ${DOXYGEN_VERSION}")
+    add_subdirectory (doxygen)
+  else ()
+    message(STATUS "Doxygen needs to be installed to generate the doxygen documentation")
+  endif ()
+endif ()
```

Comment:
There is currently no Doxygen documentation for HDF4

Reply:
Anticipating that it will slowly happen. Actually there are other files and I should leave it for later.